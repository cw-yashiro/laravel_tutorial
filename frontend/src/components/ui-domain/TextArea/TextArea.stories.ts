import { TextArea } from "./TextArea"

export default {
	title: "ui-domain/TextArea",
	component: TextArea
}

export const Default = {
	args: {}
}