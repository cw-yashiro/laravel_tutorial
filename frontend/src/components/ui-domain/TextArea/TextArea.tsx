import { forwardRef } from "react"
import style from "./style.module.css"

type Props = React.ComponentPropsWithRef<"textarea">

export const TextArea = forwardRef<HTMLTextAreaElement, Props>(
  ({ ...props }, ref) => {
    return <textarea className={style.module} {...props} ref={ref} rows={3} />
  }
)
