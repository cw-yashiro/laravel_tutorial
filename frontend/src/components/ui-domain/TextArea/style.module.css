.module {
  width: 100%;
  padding: 0.375rem 0.75rem;
  border: 1px solid #495057;
  border-radius: 0.375rem;
  margin-bottom: 0;
}
