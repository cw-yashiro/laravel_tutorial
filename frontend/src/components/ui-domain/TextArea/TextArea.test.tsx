import { describe, it, expect } from "vitest"
import { screen, render } from "@testing-library/react"
import { TextArea } from "./TextArea"

describe("ui-domain/TextArea", () => {
  it("[role='textbox']", () => {
    render(<TextArea />)
    expect(screen.getByRole("textbox")).toBeInTheDocument()
  })

  it("[role='textbox'][desabled='true']", () => {
    render(<TextArea disabled />)
    expect(screen.getByRole("textbox")).toBeDisabled()
  })

  it("Snapshot: textareaが表示される", () => {
    const { container } = render(<TextArea />)
    expect(container).toMatchSnapshot()
  })
})
