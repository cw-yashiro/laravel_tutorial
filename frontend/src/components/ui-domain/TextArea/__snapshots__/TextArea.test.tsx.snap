// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui-domain/TextArea > Snapshot: textareaが表示される 1`] = `
<div>
  <textarea
    class="_module_0efe6a"
    rows="3"
  />
</div>
`;
